### PR TITLE
Add "Show more" to long gene functions

### PR DESCRIPTION
**Pull request by dglemos**
Ticket: [G2P-312](https://www.ebi.ac.uk/panda/jira/browse/G2P-312)

### CODE REVIEW

File: src/components/curation/GeneInformation.vue [hunk 1]
```diff
@@ -1,9 +1,20 @@
 <script>
 export default {
+  data() {
+    return {
+      readMoreActivated: false,
+      MAX_CHARACTERS: 800
```
Comment (seeta-ramaraju) [inline src/components/curation/GeneInformation.vue:6]: better to keep this constant in Constants.js file and import it here

File: src/views/GeneView.vue [hunk 2]
```diff
@@ -16,6 +16,8 @@ export default {
       geneFunctionData: null,
       errorMsg: null,
       confidenceColorMap: { ...CONFIDENCE_COLOR_MAP },
+      readMoreActivated: false,
+      MAX_CHARACTERS: 800
```
Comment (seeta-ramaraju) [inline src/views/GeneView.vue:20]: better to keep this constant in Constants.js file and import it here

File: src/views/GeneView.vue [hunk 3]
```diff
@@ -16,6 +16,8 @@ export default {
       geneFunctionData: null,
       errorMsg: null,
       confidenceColorMap: { ...CONFIDENCE_COLOR_MAP },
+      readMoreActivated: false,
```
Comment (seeta-ramaraju) [inline src/views/GeneView.vue:19]:
```suggestion
      isReadMoreActivated: false,
```

File: src/components/curation/GeneInformation.vue [hunk 4]
```diff
@@ -1,9 +1,20 @@
 <script>
 export default {
+  data() {
+    return {
+      readMoreActivated: false,
```
Comment (seeta-ramaraju) [inline src/components/curation/GeneInformation.vue:5]:
```suggestion
      isReadMoreActivated: false,
```

File: src/components/curation/GeneInformation.vue (hunk 1 above)
Comment (dglemos) [reply src/components/curation/GeneInformation.vue:6]: You mean only the MAX_CHARACTERS?

Comment (seeta-ramaraju) [reply src/components/curation/GeneInformation.vue:6]: yes

File: src/components/curation/GeneInformation.vue [hunk 5]
```diff
@@ -56,7 +69,20 @@ export default {
             </div>
             <div style="width: 90%">
               <p v-if="geneFunctionData?.function?.protein_function">
-                {{ geneFunctionData.function.protein_function }} <br />
+                <span v-if="!isReadMoreActivated && geneFunctionData.function.protein_function.length > maxCharacters">
+                {{ geneFunctionData.function.protein_function.slice(0, maxCharacters) }}&hellip;
+                </span>
+                <span v-else>
+                  {{ geneFunctionData.function.protein_function }}
+                </span>
```
Comment (seeta-ramaraju) [inline src/components/curation/GeneInformation.vue:77]:
```suggestion
                <span v-if="isReadMoreActivated || geneFunctionData.function.protein_function.length <= maxCharacters">
                  {{ geneFunctionData.function.protein_function }}
                </span>
                <span v-else>
                  {{ geneFunctionData.function.protein_function.slice(0, maxCharacters) }}&hellip;
                </span>
```

Comment (seeta-ramaraju) [reply src/components/curation/GeneInformation.vue:77]: I feel this will be clearer

**Review by seeta-ramaraju — APPROVED** [review]
LGTM

**Pull request merged**